tensorflow_datasets/image/affnist.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import numpy as np
+from six.moves import urllib
+import scipy.io
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """
+@ONLINE {tijmentieleman,
+    author = "Tijmen Tieleman",
+    title  = "affNIST",
+    url    = "http://www.cs.toronto.edu/~tijmen/affNIST"
+}
+"""
+
+_DESCRIPTION = """
+The affNIST dataset for machine learning is based on the well-known MNIST dataset.
+MNIST, however, has become quite a small set, given the power of today's computers, with their multiple CPU's and sometimes GPU's.
+affNIST is made by taking images from MNIST and applying various reasonable affine transformations to them.\
+In the process, the images become 40x40 pixels large, with significant translations involved,\
+so much of the challenge for the models is to learn that a digit means the same thing in the upper right corner as it does in the lower left corner.
+"""
+
+_AFFNIST_URL = "http://www.cs.toronto.edu/~tijmen/affNIST/32x/transformed/"
+_AFFNIST_TRAIN_DATA_FILENAME = "training_batches.zip"
+_AFFNIST_VALIDATION_DATA_FILENAME = "validation_batches.zip"
+_AFFNIST_TEST_DATA_FILENAME = "test_batches.zip"
+_AFFNIST_IMAGE_SIZE = 40
+AFFNIST_IMAGE_SHAPE = (_AFFNIST_IMAGE_SIZE, _AFFNIST_IMAGE_SIZE, 1)
+AFFNIST_NUM_CLASSES = 10
+
+# _TRAIN_EXAMPLES = 1600000
+# _VALIDATION_EXAMPLES = 320000
+# _TEST_EXAMPLES = 320000
+
+
+class Affnist(tfds.core.GeneratorBasedBuilder):
+  """\
+  The affNIST dataset for machine learning is based on the well-known MNIST dataset.
+  It is made by taking images from MNIST and applying various reasonable affine transformations to them.
+  """
+  URL = _AFFNIST_URL
+
+  VERSION = tfds.core.Version(
+    "3.0.0", "New split API (https://tensorflow.org/datasets/splits)")
+  SUPPORTED_VERSIONS = [
```

Comment:
`S3 : False` version is not required in the new datasets. The new dataset version should be `0.0.1`.